### PR TITLE
refactor: DB Updates will now file level lock to stop duplicate updates

### PR DESCRIPTION
**Pull request by laf**
DO NOT DELETE THIS TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [x] Have you signed the [Contributors agreement](http://docs.librenms.org/General/Contributing/) - please do NOT submit a pull request unless you have (signing the agreement in the same pull request is fine). Your commit message for signing the agreement must appear as per the docs.
- [x] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`

I've tested `php build-base.php` and web installer both whilst running cron, validate and discovery to see if I could a) break the lock feature and/or b) have validate turn up bad schema which I couldn't on both accounts.

This is to stop the race condition we have when discovery.php could invoke a schema update when daily.sh is already doing the same. Works for me.....

Should now allow #6433 to run now without worrying about it over-running.

### CODE REVIEW

**Comment by murrant** [comment]
@laf Do you think we should include lock expiration?   That would allow us to break stuck locks eventually.

**Comment by laf** [comment]
@murrant shouldn't be difficult, get_lock() can probably do that check

**Comment by laf** [comment]
@murrant I actually think it does what you want already. So the lock isn't only checked if the file exists, it looks to see via `ps` if the pid previously recorded is still running, only if it is does the lock stay.

File: includes/functions.php [hunk 1]
```diff
@@ -2317,3 +2317,50 @@ function db_schema_is_current()
...
+    } else {
+        file_put_contents($lock_file, getmypid());
+    }
+}
+
+/**
+ * Clean up lock file for a given name
+ * @param $name
+ */
+function clean_lock($name)
```
Comment (murrant) [inline includes/functions.php]: I feel like this should be called release_lock()

Comment (laf) [inline includes/functions.php]: Agreed and done.

**Comment by murrant** [comment]
Sorry for not getting this tested sooner, but I think I'd like to push it to the next release so we can have a good lead time of testing.

**Comment by laf** [comment]
No worries on my part.

**Comment by laf** [comment]
@murrant you ok with this now?

**Comment by murrant** [comment]
Yeah, I was going to apply it and try an update then merge it, but I got busy.  If you are happy with it, you can merge.

**Pull request merged**